event-dispatcher/src/main/java/io/sk8s/event/dispatcher/EventDispatchingHandler.java

```diff
@@ -16,55 +16,264 @@
+		Topic topic = event.getResource();
+		String name = topic.getMetadata().getName();
+		topics.remove(name);
+		topicsReady.remove(name);
+	}
+
+	@EventListener
+	public void onDeploymentRegistered(ResourceAddedOrModifiedEvent<Deployment> event) {
+		Deployment deployment = event.getResource();
+		String functionName = deployment.getMetadata().getName();
+		actualNbReplicas.put(functionName, deployment.getSpec().getReplicas());
+	}
+
+	@EventListener
+	public void onDeploymentUnregistered(ResourceDeletedEvent<Deployment> event) {
+		Deployment deployment = event.getResource();
+		String functionName = deployment.getMetadata().getName();
+		actualNbReplicas.remove(functionName);
+	}
+
+	@PreDestroy
+	public synchronized void tearDown() {
+		running = false;
+		this.notify();
+		kafkaConsumerMonitor.close();
+	}
+
+	private void updateWakeUpInterval() {
+		this.scalingFrequencyMs = this.functions.isEmpty() ? 0 : 100;
+	}
+
+	private class ScalingMonitor implements Runnable {
+
+		@Override
+		public void run() {
+			while (running) {
+				synchronized (EventDispatchingHandler.this) {
+
+					Map<KafkaConsumerMonitor.TopicGroup, List<KafkaConsumerMonitor.Offsets>> offsets = kafkaConsumerMonitor
+							.compute();
+					logOffsets(offsets);
+					Map<String, Long> lags = offsets.entrySet().stream()
+							.collect(Collectors.toMap(
+									e -> e.getKey().group,
+									e -> e.getValue().stream()
+											.mapToLong(KafkaConsumerMonitor.Offsets::getLag)
+											.max().getAsLong()));
+					functions.values().stream().forEach(
+							f -> {
+								int desired = computeDesiredNbReplicas(lags, f);
+								String name = f.getMetadata().getName();
+								Integer currentDesired = actualNbReplicas.computeIfAbsent(name, k -> desired);
+								System.out.println("Want " + desired + " for " + name);
+								if (desired < currentDesired) {
+									if (edgeInfos.computeIfAbsent(name,
+											n -> System.currentTimeMillis()) < System.currentTimeMillis()
+													- DOWN_EDGE_LINGER_PERIOD) {
+										deployer.deploy(f, desired);
+										actualNbReplicas.put(name, desired);
+										edgeInfos.remove(name);
+									}
+									else {
+										System.out.printf("Waiting another %dms before scaling down to %d for %s%n",
+												DOWN_EDGE_LINGER_PERIOD
+														- (System.currentTimeMillis() - edgeInfos.get(name)),
+												desired, name);
+									}
+								}
+								else if (desired > currentDesired) {
+									deployer.deploy(f, desired);
+									actualNbReplicas.put(name, desired);
+									edgeInfos.remove(name);
+								}
+								else {
+									edgeInfos.remove(name);
+								}
+							});
+
+					try {
+						EventDispatchingHandler.this.wait(scalingFrequencyMs);
+					}
+					catch (InterruptedException e) {
+						Thread.currentThread().interrupt();
+					}
+				}
+			}
+		}
+
+		/**
+		 * Compute the desired nb of replicas for a function. Each function defines 4 values:
```

Comment:
the function doesn't yet define these 4 values (we aren't sure yet want we want to expose to users as configuration options)